exercises/queen-attack/example.exs

```diff
@@ -1,20 +1,24 @@
 defmodule Queens do
   @type t :: %Queens{white: {integer, integer}, black: {integer, integer}}
-  defstruct white: {0, 3}, black: {7, 3}
+  defstruct [:white, :black]
+  @board_range 0..7
 
   @doc """
   Creates a new set of Queens
   """
-  @spec new() :: Queens.t()
-  @spec new({integer, integer}, {integer, integer}) :: Queens.t()
-  def new(same, same), do: raise(ArgumentError)
+  @spec new(white: {integer, integer} | nil, black: {integer, integer} | nil) :: Queens.t()
+  def new(opts \\ [])
+
+  def new(white: same, black: same),
+    do: raise(ArgumentError)
+
+  def new(opts) do
+    white = opts |> Keyword.get(:white) |> check_range()
```

Comment:
Maybe can implement `check_range()` as a guard ~~> `is_valid_position()`, similar to the RobotSimulator exercise

Reply:
Not sure how to implement a guard against a function's keyword options since I can't pattern match against them (as both keys would need to be present) to extract the parameters for the guard. Or do you mean to first `Keyword.get` the options and then use the guard in a `for`/`with`?

Reply:
I mean a function which can type check that the coordinate position is valid and return a strict boolean response, it doesn't have to be placed in a `when` guard clause.